File: src/main/java/io/yody/yosurvey/survey/domain/enumeration/TargetTypeEnum.java
package io.yody.yosurvey.survey.domain.enumeration;

public enum TargetTypeEnum {
    DEPARTMENT("department", "Khảo sát phòng ban"),
    SPEC_USERS("spec_users", "Khảo sát nhân sự"),
    OTHER("other", "Khảo sát khác");

    private final String key;
    private final String description;

    TargetTypeEnum(String key, String description) {
        this.key = key;
        this.description = description;
    }

    public static boolean inValidKey(String key) {
        for (TargetTypeEnum temp : values()) {
            if (temp.key.equalsIgnoreCase(key)) return false;
        }
        return true;
    }

    public String getKey() {
        return key;
    }

    public String getDescription() {
        return description;
    }
}

File: src/main/webapp/app/shared/model/enumerations/target-type-enum.ts
export enum TargetTypeEnum {
  SPEC_USERS = 'spec_users',
  DEPARTMENT = 'department',
  OTHER = 'other',
}

export const TargetTypeLabel = {
  [TargetTypeEnum.SPEC_USERS]: 'Khảo sát nhân sự',
  [TargetTypeEnum.DEPARTMENT]: 'Khảo sát phòng ban',
  [TargetTypeEnum.OTHER]: 'Khảo sát khác',
};
